salt/metalk8s/volumes/prepared/files/metalk8s-volume-clean-up

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+import os
+import subprocess
+import sys
+
+
+def die(message):
+    print(message, file=sys.stderr)
+    sys.exit(1)
+
+
+def get_volume_id():
+    if len(sys.argv) != 2:
+        die("Invalid number of arguments: expected only one volume ID")
+
+    return sys.argv[1]
+
+
+def find_device(volume_id):
+    path_by_uuid = "/dev/disk/by-uuid/{}".format(volume_id)
+    if os.path.exists(path_by_uuid):
+        return path_by_uuid
+
+    path_by_partuuid = "/dev/disk/by-partuuid/{}".format(volume_id)
+    if not os.path.exists(path_by_partuuid):
+        die("Volume '{}' backing device was not found".format(volume_id))
+
+    partition_name = os.path.basename(os.path.realpath(path_by_partuuid))
+    device_name = os.path.basename(
+        os.path.realpath("/sys/class/block/{}/..".format(partition_name))
+    )
```

Comment:
I wonder if using regex to extract `loop[0-9]+` isn't simpler here, but maybe that way is more reliable.

Reply:
A regex would make assumptions about the naming, which is not guaranteed. Here all we assume is how sysfs is structured, which is kernel API (I believe...).

We *could* assert the final path we calculate is an actual block device (from `stat`), and a loop device (also from `stat`, though not sure how to figure out the major of `loop` except by parsing `/proc/devices`).

Reply:
(Easy using `awk`: `awk '$2=="loop" {print $1}' /proc/devices`, and likely `7` on all systems, not dynamically registered, but still, hard-coding isn't great).

Reply:
I do prefer the regex as well. It's not like we're trying to make this logic work for "any kind of device partition", we **know** it is supposed to be a loop device, so if we don't find `loop[0-9]+`, it's a good idea to bail out (if some other OS we support decides to break this behaviour, our code will explicitly tell us).

Reply:
I do agree we want to assert it's a loop-device. However, we should not do so based on some name, but based on the major of the block-device matching the major assigned to the `loop` block-device driver.

Reply:
Added a check to verify that:
- the device path exists
- it's a block device
- its major is the "loop" major (set to 7, hardcoded, sorry @NicolasT)